workflows/toast_so_sim.py

```diff
@@ -221,44 +219,77 @@ def job_create(config, jobargs, telescope, schedule, comm):
     return job, group_size, full_pointing
 
 
-def select_pixelization(job, args):
-    # Select between healpix and flat pixelization
+def select_pointing(job, args):
+    """Select the pixelization scheme for both the solver and final binning."""
+
     ops = job.operators
-    if ops.pixels_wcs_source.enabled:
-        # We are doing source mapping
-        ops.pixels_healpix_radec.enabled = False
-        ops.pixels_wcs_radec.enabled = False
-
-        # Set detector pointing
-        ops.pixels_wcs_source.detector_pointing = ops.det_pointing_azel
-        args.pixels_solve = ops.pixels_wcs_source
-        args.pixels_final = ops.pixels_wcs_source
+
+    n_enabled_solve = np.sum(
+        [
+            ops.pixels_wcs_azel.enabled,
+            ops.pixels_wcs_radec.enabled,
+            ops.pixels_healpix_radec.enabled,
+        ]
+    )
+    if n_enabled_solve != 1:
+        raise RuntimeError(
+            "Only one pixelization operator should be enabled for the solver."
+        )
+
+    n_enabled_final = np.sum(
+        [
+            ops.pixels_wcs_azel_final.enabled,
+            ops.pixels_wcs_radec_final.enabled,
+            ops.pixels_healpix_radec_final.enabled,
+        ]
+    )
+    if n_enabled_final > 1:
+        raise RuntimeError(
+            "At most, one pixelization operator can be enabled for the final binning."
+        )
+
+    # Configure Az/El and RA/DEC boresight and detector pointing and weights
+
+    ops.det_pointing_azel.boresight = ops.sim_ground.boresight_azel
+    ops.det_pointing_radec.boresight = ops.sim_ground.boresight_radec
+
+    ops.pixels_wcs_azel.detector_pointing = ops.det_pointing_azel
+    ops.pixels_wcs_radec.detector_pointing = ops.det_pointing_radec
+    ops.pixels_healpix_radec.detector_pointing = ops.det_pointing_radec
+
+    ops.pixels_wcs_azel_final.detector_pointing = ops.det_pointing_azel
+    ops.pixels_wcs_radec_final.detector_pointing = ops.det_pointing_radec
+    ops.pixels_healpix_radec_final.detector_pointing = ops.det_pointing_radec
+
+    ops.weights_azel.detector_pointing = ops.det_pointing_azel
+    ops.weights_azel.hwp_angle = ops.sim_ground.hwp_angle
+
+    ops.weights_radec.detector_pointing = ops.det_pointing_radec
+    ops.weights_radec.hwp_angle = ops.sim_ground.hwp_angle
+
+    # Select Pixelization and weights for solve and final binning
+
+    if ops.pixels_wcs_azel.enabled:
+        args.pixels_solve = ops.pixels_wcs_azel
+        args.weights_solve = ops.weights_azel
```

Comment:
I'm confused. Where is `weights_solve` added to the `args` namespace?

Reply:
This is done in the new `select_pointing()` [function here](https://github.com/simonsobs/sotodlib/blob/d052fb851f1ccd565be3c1ae1e93deb4b0d8fed4/workflows/toast_so_sim.py#L272-L281)  Many of these workflow helper functions (which deal with runtime options configuring multiple operators) can move to within the toast submodule.  However, I wanted to do that in a future, smaller PR.

Reply:
I see. Sort of like a `workflow_tools` package? I'm very much in favor of that!

I was thrown off because symbols in the `args` namespace are usually declared via `argparse` but now we are expanding it to include symbols generated as the workflow progresses. That is fine with me but it is a slippery slope into using it as a vehicle for all information flow in the workflow. Perhaps we want to call it something else than `args` if we expand its meaning?

Reply:
Ok, I see your point on abusing the `args` namespace.  I propose storing these kinds of workflow variables in the `job` namespace that we create when loading configs / options and instantiating the operators and templates.  For the high-level set of functions in the toast submodule, these should be helper functions that work on configuring many operators together.  The old "`pipeline_tools`" code was basically trying to externally configure individual operators.  Now that operators have all their configuration built-in, the kinds of things we would put in `sotodlib.toast.workflow` would be things like:

- Return a list of all nominal simulation operators and their starting defaults
- Return a list of all nominal reduction operators and their starting defaults
- Select the pointing / weights being used depending on which operators are enabled at runtime
- Function to "save" or "load" data using whichever operator combination has been enabled at runtime

The `simulate_data()` and `reduce_data()` functions in the workflows are kind of specific to the workflow, but I guess we could have a `simulate_nominal_data()` or similar in the workflow submodule.  There is a balance between reducing boilerplate / duplication in the script and hiding what the script is doing.  If a workflow has a main() that just calls 3 functions, it is really hard to follow what it is actually doing.  On the other hand, perhaps "following what is going on" is better left to interactive notebooks, and these workflows should just be about running things while reducing code duplication across scripts.

I'll fix the use of the args namespace here, and we can discuss further the workflow submodule on the next PR.

Reply:
This is now fixed